Add color matching and symmetric mirror projection with refine+preserve support

ManglerFTW opened the pull request:
This PR adds a color match option ("Match Colors to Viewport") in the Viewport Blending Settings menu that will help to match colors while generating in Refine/Restyle Texture mode.

This PR also adds a button at the bottom of the menu that will mirror the last projection on to models that a user might want to keep symmetry on. It works in Standard mode and Refine/Restyle Texture mode with Preserve Original Textures turned on.

sakalond commented:
Thanks for the contribution. I tried it and it seems to work correctly. 

Only thing I would like is if the button for MirrorReproject would be within the "Tools" section / box. Also, it would be nice if you could choose the axis for the mirroring as its now implicitly mirroring along x axis which also isn't really clear from the operator's description. If you could add this, I would be glad, but otherwise I can merge this and will do it myself.

The color matching function seems to work perfectly although I'm not sure what kind of use-cases it is good for. Is that to combat some of the color shift / saturation issues with QwenIE or IPAdapter or is it something else?

ManglerFTW commented:
I see what you mean. I tried it out but I'm still getting the seams at straight on angles when with the settings tweaked super low. 

I'm going to work on a fix for that. I've made some progress but I still need to tweak it a bit. The alpha layer doesn't seem to project right just yet. I should have some more time to work on it in a few days.

Example image showing the seam in the bottom left corner highlighted in red along with the low settings on the nodes.
<img width="2141" height="1934" alt="Screenshot_20251208_201629" src="https://github.com/user-attachments/assets/30af65eb-5130-418c-9663-f31e74d27b7c" />

sakalond commented:
Well, yes, you will get issues with straight on areas with the current implementation. You might try adjusting denoise, maybe a lower denoise will keep it more consistent, but I still think that won't fix it completely and might introduce other issues.

Also, this sort of reminds me of other area which I wanted to experiment with and that would be introducing some sort of distance based weighting for the blending as it currently only takes the angles into consideration, so if you for example had one camera pointed at a face of a character and other further away capturing the full character at the same angle, the face would get a 50/50 split between these two, which isn't ideal. Introducing a distance factor into the weight might fix that as it would still be able to do the inpainting process same way as it does for poorly covered angled surfaces. It could even solve this issue of yours without even needing to do a separate refine pass. I just feel it could be a bit finicky to get it to work right. Anyway it's one of the things I will probably try when I have some spare time.

ManglerFTW commented:
Updated mirror for X, Y, and Z axes and added controls to add a feathered vignette along the edges of the projected image along with sliders to control the thickness and softness. See the video link below for a demonstration. Should be all set to merge.

[![Watch the video](https://img.youtube.com/vi/KASQ1HC42zE/maxresdefault.jpg)](https://youtu.be/KASQ1HC42zE)
